From 11098228df9375c41866824c5f47befe12b860e7 Mon Sep 17 00:00:00 2001
From: Prathibha Madugonde <quic_prathm@quicinc.com>
Date: Mon, 8 Jul 2024 12:57:22 +0530
Subject: [PATCH BlueZ] Fix discoverable property not emitted on updating value
 to false

client/advertising.c:
Allowing discoverable property to list in the parsing
function when discoverable is off.

Test steps:
From DUT, bluetoothctl go to menu advertise
set discoverable to off and then advertise on.

Signed-off-by: Prathibha Madugonde <quic_prathm@quicinc.com>
Upstream-Status: Backport
---
 client/advertising.c | 8 +-------
 1 file changed, 1 insertion(+), 7 deletions(-)

diff --git a/client/advertising.c b/client/advertising.c
index 5c9e3eeb3..ef062a76f 100644
--- a/client/advertising.c
+++ b/client/advertising.c
@@ -404,12 +404,6 @@ static gboolean get_data(const GDBusPropertyTable *property,
 	return TRUE;
 }
 
-static gboolean discoverable_exists(const GDBusPropertyTable *property,
-							void *data)
-{
-	return ad.discoverable;
-}
-
 static gboolean get_discoverable(const GDBusPropertyTable *property,
 					DBusMessageIter *iter, void *user_data)
 {
@@ -486,7 +480,7 @@ static const GDBusPropertyTable ad_props[] = {
 	{ "ManufacturerData", "a{qv}", get_manufacturer_data, NULL,
 						manufacturer_data_exists },
 	{ "Data", "a{yv}", get_data, NULL, data_exists },
-	{ "Discoverable", "b", get_discoverable, NULL, discoverable_exists },
+	{ "Discoverable", "b", get_discoverable, NULL, NULL },
 	{ "DiscoverableTimeout", "q", get_discoverable_timeout, NULL,
 						discoverable_timeout_exits },
 	{ "Includes", "as", get_includes, NULL, includes_exists },
-- 
2.34.1

